firebase-functions/src/main/java/com/google/firebase/functions/HttpsCallableResult.kt

```diff
@@ -11,29 +11,20 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+package com.google.firebase.functions
 
-package com.google.firebase.functions;
-
-import androidx.annotation.Nullable;
-
-/** The result of calling a HttpsCallableReference function. */
-public class HttpsCallableResult {
-  // The actual result data, as generic types decoded from JSON.
-  private final Object data;
-
-  HttpsCallableResult(Object data) {
-    this.data = data;
-  }
-
+/** The result of calling a HttpsCallableReference function.  */
+class HttpsCallableResult internal constructor( // The actual result data, as generic types decoded from JSON.
+        private val data: Any?) {
   /**
    * Returns the data that was returned from the Callable HTTPS trigger.
    *
-   * <p>The data is in the form of native Java objects. For example, if your trigger returned an
+   *
+   * The data is in the form of native Java objects. For example, if your trigger returned an
    * array, this object would be a List<Object>. If your trigger returned a JavaScript object with
-   * keys and values, this object would be a Map<String, Object>.
-   */
-  @Nullable
-  public Object getData() {
-    return data;
+   * keys and values, this object would be a Map<String></String>, Object>.
+  </Object> */
+  fun getData(): Any? {
+    return data
   }
```

Comment:
What if we had a `component1()` here to [destructure](https://kotlinlang.org/docs/destructuring-declarations.html) `HttpsCallableResult`?

That way developers would call:
```kotlin
val (data) = task.result
```

instead of:
```kotlin
val data = task.result?.data
```

---

```suggestion
  }
  
  operator fun component1() = data
```

Reply:
As far as changing the API- this will probably end up getting removed entirely and replaced by the method taking a generic instead. Alternatively, this could be converted into a data class instead- to automatically get `component1()`:

```kotlin
data class HttpsCallableResult internal constructor(val data: Any?)
```

Or, even better (assuming you wanted to keep the class):
```kotlin
data class HttpsCallableResult<T> internal constructor(val data: T?)
```

Reply:
`data` classes are discouraged in our [best practices](go/fb-kotlin-best-practices#api-surface) 🥲  _(externally visible [link](https://jakewharton.com/public-api-challenges-in-kotlin/) for non-googlers)_

But I do like the idea of having a class that takes a generic :)